src/main/java/org/spongepowered/api/asset/Asset.java

```diff
@@ -77,6 +77,33 @@ default void copyToFile(Path output) throws IOException {
     }
 
     /**
+     * Copies this Asset to the specified 'outputDirectory' {@link Path}.
+     *
+     * @param outputDirectory The directory to copy to
+     * @throws IOException
+     */
+    default void copyToDirectory(Path outputDirectory) throws IOException {
+        checkNotNull(outputDirectory, "outputDirectory");
+        copyToFile(outputDirectory.resolve(this.getFileName()));
+    }
+
+    /**
+     * Returns the the last portion of the Asset URL, e.g. the file name.
+     *
+     * @return The file name
+     */
+    default String getFileName() {
```

Comment:
Since assets are not specific to a file system (that is directly supported with `Path`) there is no need to touch a filesystem-specific API here in my opinion.